### PR TITLE
Added support of ground truth images for Keras generator

### PR DESCRIPTION
**Pull request by npielawski**
Tried to add the support of ground truth images for the Keras generator. Here is an example:
`p = Augmentor.Pipeline('dataset/train/images')
p.ground_truth('dataset/train/masks')
# Augmenting operations
p.flip_left_right(probability=0.5)
p.flip_top_bottom(probability=0.5)

batch_gen = p.keras_generator
image, label = next(p.keras_generator(batch_size=1))`
where image contains the input image, and label the ground truth.

Unfortunately, this new functionality changed the API, with _execute() returning an array of images, instead of a single image.

### CODE REVIEW

File: Augmentor/Pipeline.py [hunk 1]
```diff
@@ -490,15 +525,33 @@ def keras_generator(self, batch_size, scaled=True, image_data_format="channels_l
...
+                        label = label.reshape(l, w, h)
+
+                # The X and y are populated at the given index
+                X[i] = numpy_array
+                y[i] = label
 
             if scaled:
                 X = X.astype('float32')
-                X /= 255
+                X /= 255.
```
Comment (mdbloice) [inline Augmentor/Pipeline.py]: This is something I overlooked, dividing by `255`  needs to be fixed...

File: Augmentor/Pipeline.py [hunk 2]
```diff
@@ -490,15 +525,33 @@ def keras_generator(self, batch_size, scaled=True, image_data_format="channels_l
...
+                if self.process_ground_truth_images:
+                    y = y.astype('float32')
+                    y /= 255.
```
Comment (mdbloice) [inline Augmentor/Pipeline.py]: And here of course

**Comment by mdbloice** [comment]
Hi @npielawski thanks a lot for making the PR, and sorry for the delay in replying, I have been away from my office for the past couple of weeks. Anyway, yes I think we can fix this quite easily to work with `_execute()` - however, I am also working on pretty similar functionality and will be committing that code in the next few days (I hope). Maybe then we could revisit your PR and see how to integrate what you have changed after this?

**Comment by npielawski** [comment]
Hi @mdbloice, no problem for the delay, I didn't expect anything over the summer. Please feel free to take any piece of my code to integrate in yours, if you need to. Also, I double pushed two different functionalities: the ability to work with ground truth images in Keras, but also the preallocation of the batch.

**Comment by mdbloice** [comment]
Yeah actually something I also wanted to implement was preallocation - so I will use what you've written there for that.
Thanks again for making the PR!

**Comment by npielawski** [comment]
You're welcome :)
